model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -879,71 +886,82 @@ def _create_vllm_bundle_command(
         multinode: bool,
         is_worker: bool,
         nodes_per_worker: int = 1,  # only used if multinode
+        additional_args: Optional[VLLMEndpointAdditionalArgs] = None,
     ):
         """
         VLLM start command for the single worker, or the leader in a LeaderWorkerSet.
         """
-        command = []
         subcommands = []
 
         checkpoint_path = get_checkpoint_path(model_name, checkpoint_path)
-        additional_args = infer_addition_engine_args_from_model_name(model_name)
+
+        # merge additional_args with inferred_additional_args
+        # We assume user provided additional args takes precedence over inferred args
+        vllm_args = VLLMEndpointAdditionalArgs.model_validate(
+            dict(
+                **infer_addition_engine_args_from_model_name(model_name).model_dump(
```

Comment:
think we need to do some dict1.update(dict2) to do merging of different keys, otherwise python throws a `TypeError: dict() got multiple values for keyword argument <arg>` error

Reply:
wow that's weird. 
![Screenshot 2024-10-15 at 12 05 44 PM](https://github.com/user-attachments/assets/11c6119c-d109-499d-bbef-21bddc1c1258)

Reply:
weirdly i'm getting syntax errors if try the bottom two options in our codebase

Reply:
nvmd, fixed it